newbank/server/NewBank.java

```diff
@@ -1,374 +1,536 @@
+            }
+          default:
+            return "FAIL";
+        }
+      }
     }
-
-    private String getHelpOfferLoan() {
-        return "OFFERLOAN <Amount> <FromAccount> <Terms> <intrest>\tCreates a loan for the specified period, under the defined conditions\n";
+    return "FAIL";
+  }
+
+  private String moveFundsBetweenAccounts(
+    CustomerID customerID,
+    double amount,
+    String fromAccountNumber,
+    String toAccountNumber,
+    boolean accountsBelongToSameCustomer
+  ) {
+    Customer customer = this.getCustomer(customerID);
+    Account account1 = customer.getAccount(fromAccountNumber);
+    Account account2 = accountsBelongToSameCustomer
+      ? customer.getAccount(toAccountNumber)
+      : getAccount(toAccountNumber);
+
+    if (account1 == null || account2 == null) {
+      return "Account number invalid";
     }
 
-    private String getHelpShowMyOfferedLoans() {
-        return "SHOWMYOFFEREDLOANS\t\t\t\t\t\t\t\t\tShows all offered loans of the current customer\n";
+    if (!account1.debit(amount)) {
+      return "Insufficient funds";
     }
 
-    private String getHelpShowOpenLoans() {
-        return "SHOWOPENLOANS\t\t\t\t\t\t\t\t\t\tShows all open loans with the conditions of the loan.\n";
+    account2.credit(amount);
+    return accountsBelongToSameCustomer
+      ? "Success. " +
+      amount +
+      " moved from " +
+      fromAccountNumber +
+      " to " +
+      toAccountNumber +
+      "\n\nNew Balance\n\n" +
+      account1 +
+      "\n" +
+      account2
+      : "Success. " +
+      amount +
+      " paid from " +
+      fromAccountNumber +
+      " to " +
+      toAccountNumber +
+      "\n\nNew Balance\n\n" +
+      account1;
+  }
+
+  private String createAccount(CustomerID customerId, String name) {
+    if (isValidAccountName(name)) {
+      Customer customer = this.getCustomer(customerId);
+      customer.addAccount(new Account(generateAccountNumber(), name, 0.00));
+
+      return "Account with name " + name + " was created";
+    } else {
+      return "Account not created. Illegal account name";
     }
+  }
 
-    private String getHelpAcceptLoan() {
-        return "ACCEPTLOAN <Loan Number> <Account>\t\t\t\t\tThe open loan is accepted and the amount is credited to the given account.\n";
+  private boolean isValidAccountName(String name) {
+    if (name.length() > 30) {
+      return false;
     }
 
-    private String getHelpPayBackLoan() {
-        return "PAYBACKLOAN <Loan Number>\t\t\t\t\t\t\tThe loan is repaid with interest\n";
+    for (char c : name.toCharArray()) {
+      if (Character.isDigit(c) || Character.isAlphabetic(c) || c == '-') {
+        continue;
+      } else {
+        return false;
+      }
     }
+    return true;
+  }
 
-    private String getHelpPartPayBackLoan() {
-        return "PARTPAYBACKLOAN <Loan Number> <Amount>\t\t\t\t\t\t\tThe amount is paid off the loan balance\n";
-    }
+  private boolean isValidPassword(String password) {
```

Comment:
A good method. Since we have worked with TDD throughout the project, I would suggest creating a test for this as well. The other aspects look great.